src/__tests__/index.test.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+
+import { NetworkGraph } from '..'
+
+describe('NetworkGraph', () => {
+  const mockGraphProps = {
+    data: {
+      nodes: [
+        { id: 'node1', label: 'Node 1' },
+        { id: 'node2', label: 'Node 2' },
+        { id: 'node3', label: 'Node 3' },
+      ],
+      links: [
+        { source: 'node1', target: 'node2' },
+        { source: 'node2', target: 'node3' },
+      ],
+    },
+  }
+
+  it('renders the graph component', () => {
```

Comment:
Let's assert the following in this file:

- The graph renders with the right number of nodes and links (for example: assert manually getAllByTestId('node').length === X number)
- The graph renders the default node and default link if not provided 
- The graph renders custom nodes and links (in custom nodes pass prop `selectNode`
        - assert that when a node is clicked, the function is called)
        - assert that the node does something when the prop isActive is true, so you validate that that is working as expected
- pass the prop `hoverOpacity` and assert that when hovering over a node, the opacity actually changes
- Pass a basic DetailsComponent and validate it is rendered when clicking over a node, and that it uses the props `node: Node, unselectNode(): void` as expected (rendering the name of the node, and a button that when clicked, executes `unselectNode`

With all these cases we are more than covered with all important scenarios

Reply:
All tests were added, but I commented test about `hoverOpacity`, because any event from `fireEvent` didn't reproduce `hoverOpacity` behaviour. I also tried to use "Force element state" in chrome inspector, and it also didn't reproduce `hoverOpacity` behaviour. I checked code, and it triggers by `mouseover` event, but still, `fireEvent.mouseover` doesn't work it that case.